### ISSUE TITLE
escpod cannot run a leech waveform charging model: move the chunk assembly into escapepod-signal, then add the bundle variant

### ISSUE THREAD

**Issue by jayhesselberth** [rnabioco/escapepod-rs#306]
`escapepod-models` now ships a charging model this runtime cannot load:
**`charging_tcn_rna004@v0.1.0`**, a leech waveform network (`TCNDwellResidualLN`)
that beats the shipped feature network by **+0.0050 AUROC** on the fnn's own
training rows and test reads (n=1,387,667; 3 seeds x 2 geometries, 6/6 positive,
paired sd 0.00008, McNemar z=-136; 0.988 recall against 0.960 at the fnn's own
FPR). It is in the registry and refused at the door, which is the same state
`charging_feature_nn_rna004` shipped in before #231.

This issue is the work to make `escpod signal classify` run it.

## Why it is refused today

1. **No variant.** `bundle.rs`'s `VariantProbe` knows `gbm` and `feature_model`,
   both **one input tensor and a two-class softmax**. This model takes **three
   tensors and emits one BCE logit**, and the bundle declares it under a new
   `waveform_model` key.
2. **No refinement.** The corpus was built with the banded DP on (`scale_iters
   2`, `half_bandwidth 5`), so the base-to-signal map the features are taken
   over is not the move table's. `bundle.rs:889` already refuses the *other*
   variants for exactly this ("its spans come straight from the move table, so
   every feature would be taken over a different stretch of signal") — the same
   sentence is the blocker here, not just a schema gap.

## The decision (rnabioco/escapepod-models, this session)

The chunk assembly **moves from leech-core down into `escapepod-signal`**, and
both consumers call it: leech's Rust pipeline and `escapepod-classify`. Not a
third private copy in the consumer — this rule already has two implementations
(leech Python, leech-core Rust) and the failure mode is on the record:
escapepod-classify reproduced a superseded feature definition for two months,
and its counted golden missed it because all 19 fixture reads took the other
branch.

Direction is the one already established by leech#200/#202/#203, which handed
the per-base stats, the k-mer levels and the refinement preset down to
`escapepod-signal`: leech-core depends on escapepod-signal, never the reverse.

## What already exists here (with leech parity tests)

* `escapepod_signal::seq_encoding::{encode_signal_kmer, sequence_to_int}` —
  `tests/signal_kmer_parity.rs`
* `escapepod_signal::resquiggle::{extract_levels, load_kmer_table}` —
  `tests/kmer_levels_parity.rs`
* `escapepod_signal::features::span_stats` (`SpanConfig`, `Normalization`,
  `SpanFill`, `SpanBounds`, `MedianConvention`) — **leech-core's
  `compute_per_base_stats` already calls it**
* the refinement preset — `tests/test_refine_preset.rs`

So the primitives are done. What has to move is the **assembly**.

## What has to move (leech-core -> escapepod-signal)

Reference implementation: `rnabioco/leech`, `rust/src/inference_pipeline/`:
`processing.rs` (`process_read_signal`), `signal_mapping.rs`
(`compute_ref_to_signal`, `build_seq_to_sig_map`, `chunk_signal_kmer_inputs`),
`features.rs` (`compute_dwell_features`), `refinement.rs`, `types.rs`
(`PipelineConfig`, `BaseJustify`), `inference.rs` (`extract_inference_chunks`).

Keep it **generic** — anchor, window, channels. No assay vocabulary in a signal
crate (that rule has been broken twice in escapepod-signal and corrected twice).

### The contract this model needs

Three `f32` inputs, batch-major:

| input | shape | what |
|---|---|---|
| `signal` | `[B, 2, 390]` | ch0 normalised current, ch1 its k-mer residual |
| `sequence` | `[B, 36, 390]` | `encode_signal_kmer` of the 11-mer context, scattered along the signal axis: `4 * (4 + 4 + 1) = 36` rows |
| `features` | `[B, 12, 21]` | per-base rows over reference offsets `0..=20` |

Output `[B, 1]`: **a single BCE logit whose positive class is `uncharged`**, so
`P(charged) = 1 - sigmoid(logit)` and `cl = round(P(charged) * 255)`. Read as
P(charged) — the obvious reading, and what every other variant emits — every
call inverts and nothing errors. leech assigns class ints at merge time and gave
`charged` 0 on this corpus; the bundle states this under
`waveform_model.output`.

**The 12 feature rows, in order — this order IS the trained input:**

```
0 dwell            5 level_mean         9  kmer_expected
1 dwell_log        6 level_median       10 kmer_residual
2 dwell_mean       7 level_std          11 kmer_residual_abs
3 dwell_std        8 level_range
4 dwell_ratio
```

i.e. `compute_dwell_features` order, then `compute_signal_features` order, then
the three k-mer residual rows `preparation/reader.py` folds into the signal
dict. `leech/chunking/extractor.py::merge_feature_channels` pins it and says
plainly that reordering "silently feeds `level_mean` into the filter that
learned `dwell_log`". `dwell_mean`/`dwell_std` are a **5-wide centred rolling
window with edge padding**; `dwell_log = ln(d + 1e-6)`; `dwell_ratio =
d / (dwell_mean + 1e-6)`.

### Geometry, from the bundle rather than from constants

`models/charging_tcn_rna004@v0.1.0/metadata.json`, under
`waveform_model.preprocessing` (copied verbatim from the corpus's own
`prepare_config.json`, so it cannot drift from what training did):

```
anchor            reference, motif CCAGGC, motif_offset 2   <- NOT +3
base_justify      end
signal_context    [90, 300]  ->  signal_len 390
reverse_signal    true       (5'->3' before the window is cut)
signal_norm       median_mad (per read)
feature_start/end 0 / 20     (21 bases)
kmer_len          11         signal_kmer_context [4, 4]
refine_signal_map true, scale_iters 2, half_bandwidth 5, kmer_center_idx 4
kmer_table        9mer_levels_v1.txt, sha256 in the bundle and travelling with it
```

Note `motif_offset` **2**, against the feature-grid variants' `+3`. Inheriting
the other variant's offset places every window three bases off and validates
cleanly.

## Then in escapepod-classify

* `VariantProbe` learns a third arm keyed on `waveform_model` (exactly one of
  `gbm` / `feature_model` / `waveform_model`, as today).
* Wire `escapepod-signal`'s refinement into the classify feature path, driven by
  the bundle's declared parameters, and narrow the `bundle.rs:889` refusal to
  the variants that still cannot do it.
* Run the graph. **Open question, answer it first:** `fnn.rs` uses `tract`; the
  demux CRF path uses `ort`. This graph is a dilated TCN with LayerNorm and an
  `adaptive_avg_pool1d` that already forced the *dynamo* exporter (the legacy
  TorchScript path fails on it), so whether tract handles it at opset 18 is
  unknown and is the first thing to test — load
  `models/charging_tcn_rna004@v0.1.0/charging_tcn_rna004.onnx` and see. If tract
  cannot, use `ort` behind a feature flag as the CRF path does, and say so in
  the bundle's runtime note.
* Emit `cl` from `1 - sigmoid(logit)`, and keep the abstain semantics the bundle
  declares (`no chunk, no call` — this variant has no
  `aligner_arm_depth` rule and must not inherit one).

## Acceptance

1. **Parity, not just "it runs".** Per-read `cl` from `escpod signal classify`
   against the Python scores on the same reads. The fnn's bar was max |dP|
   6.0e-05 on real reads; the ONNX itself already round-trips at 1.4e-05 max on
   the logit with **0 decision disagreements over 4,096 real chunks**
   (`results/charging_tcn_rna004/onnx_verify.json` in escapepod-models), so any
   gap beyond that is the tensor assembly, not the graph.
2. **Goldens that exercise both branches**, not only the majority path — the
   two-month divergence above was invisible because 19 fixture reads all took
   one branch. Include reads with an unmapped junction and reads whose window
   runs off the end.
3. **leech-core calls the moved code and its copy is deleted**, with leech's own
   tests still bit-identical (`tests/test_data_prep.py::TestFeatureChannelOrder`
   pins the row order). Two implementations left standing is the thing this
   issue exists to prevent.
4. `escapepod-models`'s `tests/test_charging_bundle.py::
   test_leech_bundle_is_refused_by_the_runtime` **flips** — it currently asserts
   the refusal, so it is the tripwire that says when this landed.

## Artifacts to work against

* Bundle: `escapepod-models` branch `ship-charging-tcn`,
  `models/charging_tcn_rna004@v0.1.0/` (metadata + the self-contained ONNX +
  the k-mer table; the ONNX is gitignored and ships on the Release).
* A corpus that reproduces the inputs exactly:
  `benchmarks/ldx16x/charging_leech/production/merged/test/test.npz` plus its
  `prepare_config.json`, on the same branch.
* Per-read reference scores: the arm's `test_scores.npz`
  (`read_ids`, `probs` = P(uncharged)) under
  `benchmarks/ldx16x/charging_leech/models/production/TCNDwellResidualLN/`.

Tracked on the model side as rnabioco/escapepod-models#61.

**Comment by jayhesselberth** [comment]
**The bundle now declares its own input contract**, so this work reads the artifact rather than this issue's prose. Two gaps the model's author (rightly) caught after filing:

* **Row names.** `waveform_model.channels.features.order` carries all twelve, in the trained order — `dwell, dwell_log, dwell_mean, dwell_std, dwell_ratio, level_mean, level_median, level_std, level_range, kmer_expected, kmer_residual, kmer_residual_abs` — plus one-line definitions, and `channels.signal.order` = `[signal, signal_kmer_residual]`. It is **derived from leech at build time** (`merge_feature_channels` over the two dicts) and cross-checked against the checkpoint's `num_features`, so a reordering upstream fails the model build instead of shipping a mislabelled tensor. It is not transcribed anywhere.

* **The placement rule.** `anchor.motif` = CCAGGC, `anchor.motif_offset` = **2** (not the feature-grid variants' +3), `preprocessing.motif_reference` = `fasta` (the motif is found in the REFERENCE), and `preprocessing.base_justify` = `end` — which means the focus is the anchor base's span **END**, not its midpoint. `preprocessing.focus_rule` spells it out: *find CCAGGC in the reference, take the base at +2, map it through the (refined) base-to-signal map, focus = that base's span end, window = [focus - 90, focus + 300)*. Centring on the midpoint instead is ~half a base — about 18 samples here — on every read, with nothing to signal it.

The whole 27-key `prepare_config.json` also rides along verbatim under `preprocessing.prepare_config`, so knobs nobody thought to surface (`skip_motif_indels`, `recover_softclip_signal`, the reference fasta) are in the bundle rather than only in a corpus directory that may be purged.

Both are pinned by tests in escapepod-models (`test_leech_bundle_names_every_tensor_row`, `test_leech_bundle_ships_the_placement_rule`), so they cannot quietly drop out of a later build.
